### PR TITLE
libsForQt5.qtbase: fix Darwin build by removing obsolete patches

### PR DESCRIPTION
**Pull request by emilazy**
Another existing bug exposed by 514b00cf08702b31cdf873a798f1ff100d4f2cf7: the build system was interpolating an empty SDK directory and passing `-isysroot  -g0`, eating a compiler flag and breaking the build now that we don’t pass a redundant `-isysroot` of our own.

Thankfully, with the new SDK pattern, we don’t need to do any of this elaborate patching; just pointing it at the right tools from `xcbuild` lets it figure everything out itself.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin (on a branch with some other changes, though)
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/qt-5/modules/qtbase.nix [hunk 1]
```diff
@@ -146,13 +149,26 @@ stdenv.mkDerivation (finalAttrs: ({
...
+        configure \
+        mkspecs/features/mac/asset_catalogs.prf \
+        mkspecs/features/mac/default_pre.prf \
+        mkspecs/features/mac/sdk.mk \
+        mkspecs/features/mac/sdk.prf
+      do
+        substituteInPlace "$file" \
+          --replace-quiet '/usr/bin/xcode-select' '${lib.getBin xcbuild}/bin/xcode-select' \
+          --replace-quiet '/usr/bin/xcrun' '${lib.getBin xcbuild}/bin/xcrun' \
+          --replace-quiet '/usr/libexec/PlistBuddy' '${lib.getBin xcbuild}/bin/PlistBuddy'
```
Comment (reckenrode) [inline pkgs/development/libraries/qt-5/modules/qtbase.nix:162]: Would it be better to use `lib.getExe'`?

Comment (emilazy) [reply pkgs/development/libraries/qt-5/modules/qtbase.nix]: Sure, why not. (Pushed some miscellaneous other fixes and building again.)

**Comment by emilazy** [comment]
This builds successfully on my `staging` branch and the `otool -l` output looks right; only difference vs. `staging-next` is that `libQt5Core.dylib` links against `/usr/lib/libobjc.A.dylib` for some reason, but that should be harmless and might even be more correct. This should be ready to go.

**Comment by K900** [comment]
ack

**Comment by emilazy** [comment]
Built `libsForQt5.{qtbase,qttools,qttranslations,qtimageformats,qtdeclarative,qtmultimedia}` and the `otool -l` output all looks good.

**Pull request merged**